src/raftstore/store/snap.rs

```diff
@@ -941,7 +959,11 @@ impl Snapshot for Snap {
     }
 
     fn apply(&mut self, options: ApplyOptions) -> Result<()> {
-        box_try!(self.validate());
+        if self.validate().is_err() {
+            // if checksum failed, we do a lazy sst seq no modification
+            box_try!(self.set_default_sst_seq_no(&options));
+            box_try!(self.validate());
+        }
```

Comment:
Assuming:
1) ingest sst for default cf success(using hard link mode instead of moving), let's assuming the file number inside rocksdb is 00001.
2) and then the machine crash down.
3) after restart, reset the global sequence number of sst and ingest it again. When ingesting, RocksDB will modify the sst's global sequence number.
4) if there is a compaction which including sst.00001 is running,  and 3) will change its global sequence number, what will happen?

Reply:
@huachaohuang

Reply:
What will happen, if we change the global sequence number of a sst file which belongs to RocksDB, will it break the consistency of reading?

Reply:
To avoid these conner cases, when first validation failed, we should copy these files out and modify.

Reply:
Well, seems we're not supposed to modify an ingested file.

Reply:
IMO, the whole idea seems too hacky, we should look for a better solution.

Reply:
Yes, this is hacky, and another way to do this is let raft send a new snapshot.
Do you have any idea?

Reply:
Not yet.